### DISCUSSION TITLE
Intermittent Errors During Workflow Execution

### DISCUSSION THREAD

**Discussion by Mike-E-angelo** [community/community#25058] [Actions]
<p>I am encountering errors during executions which seem to go away when re-running the workflow.  Here is an example:</p>
<p><a href="https://github.com/Mike-EEE/action-jackson/runs/320906130" rel="nofollow noopener">https://github.com/Mike-EEE/action-jackson/runs/320906130</a></p>
<p>The error message seems to have a problem with this line:</p>
<p><a href="https://github.com/Mike-EEE/action-jackson/blob/e72becabd0ef9d23dd951ddcb8d022deda39ffcf/.github/workflows/changelog-dawg.yml#L24" rel="nofollow noopener">https://github.com/Mike-EEE/action-jackson/blob/e72becabd0ef9d23dd951ddcb8d022deda39ffcf/.github/workflows/changelog-dawg.yml#L24</a></p>
<p>But as you can see in the output task in the previous step, the expected field is emitted as an output value:</p>
<pre><code>::set-output name=filename::CHANGELOG.md
</code></pre>
<p>Re-running tests seem to make this go away (although I have encountered failures on subsequent retries), but I thought I would like to report this as it *seems* everything is configured correctly on my side.</p>
<p>I am, of course, open to any suggestions on anything I could be doing incorrectly here.  Thank you for any guidance and/or consideration.</p>

**Comment by weide-zhou** [comment]
<p>I forked your repo and the issue doesn’t repro, and it seems the error is gone on your recent workflows, so it’s fine now, right?</p>
<p>I checked the error message, looks the path is not correctly created or accessed, please check the similar <a href="https://github.com/actions/upload-artifact/issues/13" rel="nofollow noopener">ticket</a>, the path should be checked and make sure it’s available in the $GITHUB_WORKSPACE.</p>

**Comment by Mike-E-angelo** [comment]
<p>Hi <a class="mention" href="/weide-zhou">@weide-zhou</a> thank you for your reply.  Sorry for the delay here as “subscribe” is not enabled by default and I had to visit the site to see if there were any updates.</p>
<p>I’m afraid this happens quite regularly still… especially when one attempts to push a release as we are now. :sweat_smile:</p>
<p>I have had this occur on three repositories now, all of them intermittent, and all go away after retrying:</p>
<p><a href="https://github.com/Mike-EEE/action-jackson/blob/71937e8e44c08614692a7f7f1bb5ba8936e105e5/.github/workflows/publish-release-build.yml#L28" rel="nofollow noopener">https://github.com/Mike-EEE/action-jackson/blob/71937e8e44c08614692a7f7f1bb5ba8936e105e5/.github/workflows/publish-release-build.yml#L28</a></p>
<p><a href="https://github.com/ExtendedXmlSerializer/NextRelease/blob/d7176e58f2964b793cdf7210d5a4667677047532/.github/workflows/publish-release-build.yml#L28" rel="nofollow noopener">https://github.com/ExtendedXmlSerializer/NextRelease/blob/d7176e58f2964b793cdf7210d5a4667677047532/.github/workflows/publish-release-build.yml#L28</a></p>
<p><a href="https://github.com/ExtendedXmlSerializer/home/blob/47f7567de9a45f849ffa106bd391d11026b8a2eb/.github/workflows/publish-release-build.yml#L28" rel="nofollow noopener">https://github.com/ExtendedXmlSerializer/home/blob/47f7567de9a45f849ffa106bd391d11026b8a2eb/.github/workflows/publish-release-build.yml#L28</a></p>
<p>As for checking, $GITHUB_WORKSPACE: why would this work in some runs and not others?  Any guidance on how to check for this would be appreciated, too… are you saying we have to use script here for this particular value?</p>
<p>Again “re-run all checks” fixes the problem.  Usually this fixes it on the 2nd attempt, but I have had to retry up to four times to make it work.</p>
<p>Finally, I did check that ticket you provided (thank you!) but this seems to be a different issue.  My guess is that the template engine is not evaulating the <code>${{&amp;nbsp;steps.generate.outputs.filename }}</code> in time and it is resulting in the <code>null</code> value presented in the error.</p>

**Comment by Mike-E-angelo** [comment]
<p>Encountered this again.  here is the error in full:</p>
<pre><code>2019-12-03T07:16:02.2408579Z ##[error]Value cannot be null. (Parameter 'path')
2019-12-03T07:16:02.2719818Z ##[error]Exit code 1 returned from process: file name '/home/runner/runners/2.161.0/bin/Runner.PluginHost', arguments 'action "GitHub.Runner.Plugins.Artifact.PublishArtifact, Runner.Plugins"'.
</code></pre>
<p>So it seems in this case, the <code>path</code> value is <code>null</code> which makes me suspect that the templated value <code>${{&amp;nbsp;steps.generate.outputs.filename }}</code> is not being evaluated in time.  Again, re-running checks eventually make it work, so it eventually evaulates in time.  However, there are times (which seem more often when trying to actually deploy rather than test :sweat_smile:) when it doesn’t evaluate in time, leading to the error.  This is my hunch, however.</p>

**Comment by Mike-E-angelo** [comment]
<p>Now it would seem that I am consistently <a href="https://github.community/t/failed-to-cancel-check-suite/16355/9">running into this issue</a> and cannot re-run check suite.  Or I can but the check suite does not actually re-run and I cannot cancel it. :disappointed:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/7/7df56e0bbc1acedea2d5db2336f36995c7b9d73c.png" data-download-href="/uploads/short-url/hYhvV9XZ9uBhcCuiWlrUYdYW6Ze.png?dl=1" title="qKRkwa7" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/2931376/181088601-794ba56b-2941-4cf4-8991-41f524704151.png" alt="qKRkwa7" data-base62-sha1="hYhvV9XZ9uBhcCuiWlrUYdYW6Ze" width="690" height="322" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/7/7df56e0bbc1acedea2d5db2336f36995c7b9d73c_2_690x322.png, https://aws1.discourse-cdn.com/github/optimized/2X/7/7df56e0bbc1acedea2d5db2336f36995c7b9d73c_2_1035x483.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/7/7df56e0bbc1acedea2d5db2336f36995c7b9d73c_2_1380x644.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/7/7df56e0bbc1acedea2d5db2336f36995c7b9d73c_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">qKRkwa7</span><span class="informations">1548×723 59 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Seen here:</p>
<p><a href="https://github.com/ExtendedXmlSerializer/home/actions" rel="nofollow noopener">https://github.com/ExtendedXmlSerializer/home/actions</a></p>

**Comment by weide-zhou** [comment]
<p>Sorry for the inconvenience taken,  please ask private <a href="https://support.github.com/contact" rel="nofollow noopener">support</a> to help handle the hanging workflow runs.</p>

**Comment by Mike-E-angelo** [comment]
<p>Ok cool!  Thank you <a class="mention" href="/weide-zhou">@weide-zhou</a> I did not know that was possible. :slight_smile:</p>
<p>What about the intermittent errors?  I just now ran into another one.  Creating releases is nearly impossible with the current GitHub Actions because of this.</p>
<p>Again the issue does not seem to be in the action, but in the evaluated parameters that are sent to it.  From what I can tell the action never actually runs, and it is still in the process of validating the parameters sent to it.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/2/2b0246bd0ca8722455078ab8ff54f024420f66c2.png" data-download-href="/uploads/short-url/68tpjdW9tx9LZ0GbadMIemW2RzA.png?dl=1" title="WoCfGXG" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/2931376/181088604-f853f20f-b18c-43a9-b53b-aacaa3336b48.png" alt="WoCfGXG" data-base62-sha1="68tpjdW9tx9LZ0GbadMIemW2RzA" width="690" height="262" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/2/2b0246bd0ca8722455078ab8ff54f024420f66c2_2_690x262.png, https://aws1.discourse-cdn.com/github/optimized/2X/2/2b0246bd0ca8722455078ab8ff54f024420f66c2_2_1035x393.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/2/2b0246bd0ca8722455078ab8ff54f024420f66c2_2_1380x524.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/2/2b0246bd0ca8722455078ab8ff54f024420f66c2_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">WoCfGXG</span><span class="informations">1586×603 37.7 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by weide-zhou** [comment, marked as answer]
<p>Sorry for late response <a class="mention" href="/mike-e-wins">@mike-e-wins</a> ! </p>
<p>The runner doesn’t parse the ‘::’ in the middle successfully. <a href="https://github.com/DragonSpark/action-changelog-dawg/blob/master/entrypoint.ps1#L39" rel="nofollow noopener">https://github.com/DragonSpark/action-changelog-dawg/blob/master/entrypoint.ps1#L39</a></p>
<p>Please use </p>
<pre><code>echo "`n::set-output name=filename::$filename"
</code></pre>
<p>instead. It works on my end.</p>
<p><img src="https://user-images.githubusercontent.com/52772761/181088606-fe109259-d6b5-47e5-97e9-cf8504814c8f.png" alt="intermitenterror.png" data-base62-sha1="xmNYqOnDG0Pi8Si5p3Az3KvGRgM" width="659" height="264"></p>

**Comment by Mike-E-angelo** [comment]
<p>Cool!  Thanks for looking into this <a class="mention" href="/weide-zhou">@weide-zhou</a>.  I will see how this does and let you know. :+1:</p>

**Comment by Mike-E-angelo** [comment]
<p>That seemed to do the trick <a class="mention" href="/weide-zhou">@weide-zhou</a> !  Thank you for your efforts and taking the time to improve your product.  GitHub Actions is quite cool. :slight_smile:</p>
